### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by iamsjunaid**
# In this milestone of the project, I:

- Created the following tables:
    - [x] `vets`

- Created the following tables having many-to-many relationships:
    - [x] `specializations` (`species` and `vets`)
    - [x] `visits` (`animals` and `vets`)

- Inserted the data in the following tables:
    - [x] `vets`
    - [x] `specializations`
    - [x] `visits`
 
- Wrote the queries for the following:
    - [x] Who was the last animal seen by William Tatcher?
     
    
    - [x] How many different animals did Stephanie Mendez see?
     
   
    - [x] List all vets and their specialties, including vets with no specialties.
     

    - [x] List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
     

    - [x] What animal has the most visits to vets?
     

    - [x] Who was Maisy Smith's first visit?
     

    - [x] Details for most recent visit: animal information, vet information, and date of visit.
     

    - [x] What specialty should Maisy Smith consider getting? Look for the species she gets the most. 
     
    - [x] How many visits were with a vet that did not specialize in that animal's species?

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -279,5 +279,121 @@ GROUP BY
...
+  JOIN animals ON visits.animal_id = animals.id
+  JOIN vets ON visits.vet_id = vets.id
+  JOIN specializations ON vets.id = specializations.vet_id
+  JOIN species ON specializations.species_id = species.id
+WHERE
+  vets.name = 'Maisy Smith'
+GROUP BY
+  species.name
+ORDER BY
+  COUNT(*) DESC
```
Comment (Zuhaib042) [inline queries.sql]: - [x] Kindly consider correcting this query as this results in a wrong result. Maisy Smith should consider getting `Digimon` as she got 6 visits of that species.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @iamsjunaid 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Well written queries
✅ Clean code

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by iamsjunaid** [comment]
- [x] What specialty should Maisy Smith consider getting? Look for the species she gets the most.
    image

File: queries.sql (hunk 1 above)
Comment (iamsjunaid) [reply queries.sql]: Hi @Zuhaib042, Thanks for bringing this to my attentions😊.  I have updated the query.

**Review by Ismailco — APPROVED** [review]
### Hi @iamsjunaid, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**